renderer/canvas.py

```diff
@@ -221,6 +231,154 @@ def _draw_container_image(self, surface, container_name, obs, position):
+                sprite = spritesheet.subsurface(rect).convert_alpha()
+                sprites.append(sprite)
+        # Add empty sprite for index -1
+        empty = pygame.Surface((1, 1), flags=pygame.SRCALPHA)
+        empty.fill((0, 0, 0, 0))
+        sprites.append(empty)
+                
+        # Parse tile ID mappings
+        mappings = {}
+        for word, walls in tiling_config["mappings"].items():
+            entry = {}
+            for data in walls:
+                wall = data[0]
+                corners = {}
+                for t in data[1]:
+                    corners[t[0]] = t[1]
+                entry[wall] = corners
+            mappings[word] = entry
+
+        # Store loaded values in catalog
+        catalog = {}
+        catalog["config"] = tiling_config
+        catalog["sprites"] = sprites
+        catalog["mappings"] = mappings
+        return catalog
+        #self.asset_directory[tiling] = catalog
+    
+    def _parse_abstract_tile_matrix(self, abstract_matrix, tiling_catalog):
+        """
+        Parse the given abstract tile matrix into a matrix of raw tile IDs suitable for drawing
+
+        Args:
+            abstract_matrix (List[String]): List of strings whose characters represent abstract tiles
+            tiling_catalog (Dict): Catalog to reference when translating tilings
+        """
+        raw_matrix = [[-1] * len(abstract_matrix[0]) for _ in range(len(abstract_matrix))]
+        mappings = tiling_catalog["mappings"]
+
+        for row in range(len(abstract_matrix)):
+            for column in range(len(abstract_matrix[row])):
+                letter = abstract_matrix[row][column]
+                if not letter in mappings:
+                    raw_matrix[row][column] = -1
+                    continue
+
+                N_wall = "1" if row == 0 or abstract_matrix[row - 1][column] != letter else "0"
+                S_wall = "1" if row == len(abstract_matrix) - 1 or abstract_matrix[row + 1][column] != letter else "0"
+                W_wall = "1" if column == 0 or abstract_matrix[row][column - 1] != letter else "0"
+                E_wall = "1" if column == len(abstract_matrix[row]) - 1 or abstract_matrix[row][column + 1] != letter else "0"
+
+                NE_corner = "1" if row == 0 or column == len(abstract_matrix[row]) - 1 or abstract_matrix[row - 1][column + 1] != letter else "0"
+                SE_corner = "1" if row == len(abstract_matrix) - 1 or column == len(abstract_matrix[row]) - 1 or abstract_matrix[row + 1][column + 1] != letter else "0"
+                NW_corner = "1" if row == 0 or column == 0 or abstract_matrix[row - 1][column - 1] != letter else "0"
+                SW_corner = "1" if row == len(abstract_matrix) - 1 or column == 0 or abstract_matrix[row + 1][column - 1] != letter else "0"
+
+                # corners covered by a wall are redundant
+                if N_wall == "1":
```

Comment:
I see what the comment was for now. yes this is good to inform a user about since it is ambiguous whether or not it is necessary to fill in the corner information (renderer will ignore it)

Reply:
one concern here is whether this code will ignore the corners in the mapping. it looks like right now it is assumed that the user also inputs the information correctly.